src/decoders/ascii85_decoder.rs

```diff
@@ -0,0 +1,176 @@
+                decoded_text
+            );
+            return results;
+        }
+
+        let checker_result = checker.check(&decoded_text);
+        results.unencrypted_text = Some(vec![decoded_text]);
+
+        results.update_checker(&checker_result);
+
+        results
+    }
+    /// Gets all tags for this decoder
+    fn get_tags(&self) -> &Vec<&str> {
+        &self.tags
+    }
+    /// Gets the name for the current decoder
+    fn get_name(&self) -> &str {
+        self.name
+    }
+}
+
+/// helper function
+fn decode_ascii85_no_error_handling(text: &str) -> Option<String> {
+    // Runs the code to decode ascii85
+    // Doesn't perform error handling, call from_ascii85
+    ascii85::decode(text)
+        .ok()
+        .map(|inner| String::from_utf8(inner).ok())?
+}
+
+#[cfg(test)]
+mod tests {
+    use super::ASCII85Decoder;
+    use crate::{
+        checkers::{
+            athena::Athena,
+            checker_type::{Check, Checker},
+            CheckerTypes,
+        },
+        decoders::interface::{Crack, Decoder},
+    };
+
+    // helper for tests
+    fn get_athena_checker() -> CheckerTypes {
+        let athena_checker = Checker::<Athena>::new();
+        CheckerTypes::CheckAthena(athena_checker)
+    }
+
+    #[test]
+    fn successful_decoding() {
+        let ascii85_decoder = Decoder::<ASCII85Decoder>::new();
+
+        let result = ascii85_decoder.crack("BOu!rD]j7BEbo7", &get_athena_checker());
+        let decoded_str = &result
+            .unencrypted_text
+            .expect("No unencrypted text for ascii85");
+        assert_eq!(decoded_str[0], "hello world");
+    }
+
+    #[test]
+    fn ascii85_decode_empty_string() {
+        // Bsae64 returns an empty string, this is a valid ascii85 string
+        // but returns False on check_string_success
+        let ascii85_decoder = Decoder::<ASCII85Decoder>::new();
+        let result = ascii85_decoder
+            .crack("", &get_athena_checker())
+            .unencrypted_text;
+        assert!(result.is_none());
+    }
+
+    #[test]
+    fn ascii85_decode_handles_panics() {
+        let ascii85_decoder = Decoder::<ASCII85Decoder>::new();
+        let result = ascii85_decoder
+            .crack(
+                "hello my name is panicky mc panic face!",
+                &get_athena_checker(),
+            )
+            .unencrypted_text;
+        if result.is_some() {
+            panic!("Decode_ascii85 did not return an option with Some<t>.")
+        } else {
+            // If we get here, the test passed
+            // Because the ascii85_decoder.crack function returned None
+            // as it should do for the input
+            assert_eq!(true, true);
+        }
+    }
+
+    #[test]
+    fn ascii85_handle_panic_if_empty_string() {
+        let ascii85_decoder = Decoder::<ASCII85Decoder>::new();
+        let result = ascii85_decoder
+            .crack("", &get_athena_checker())
+            .unencrypted_text;
+        if result.is_some() {
```

Comment:
I don't know which decoder you used as a template but these tests were updated/improved. (panic, empty string, and emoji tests). Use the base65536 decoder as an example.